src/mvp/commontypes.hpp

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <map>
+#include <stdint.h>
+
+const int GridCells = 10;
+
+struct NetworkMessageTypes {
+    static const uint16_t PlayerReady = 0;
+    static const uint16_t ReadyConfirmation = 1;
+    static const uint16_t GameStart = 2;
+    static const uint16_t GameStop = 3;
+    static const uint16_t BuyShip = 4;
+    static const uint16_t MoveShip = 5;
+    static const uint16_t AttackShip = 6;
+    static const uint16_t BoardUpdate = 7;
+};
+
+struct ShipType {
+    static const uint8_t None = 0;
+    static const uint8_t Cruiser = 1;
+    static const uint8_t Destroyer = 2;
+};
+
+struct ShipAction {
+    static const uint8_t None = 0;
+    static const uint8_t Move = 1;
+    static const uint8_t Attack = 2;
+};
+
+struct ShipInfo {
+    uint16_t Damage;
+    uint16_t Health;
+    uint16_t Cost;
+    uint16_t textureOffsetX;
+    uint16_t textureOffsetY;
+};
```

Comment:
Why are these structs with static data instead of enums that inherit from a type? Like:

```c++
struct NetworkMessageTypes : uint16_t {
    PlayerReady = 0;
    ReadyConfirmation;
    GameStart;
    // ...
}
```